src/python/pants/engine/target.py

```diff
@@ -642,6 +649,23 @@ def validate(self) -> None:
         """
 
 
+def _validate_origin_text_blocks(origin_text_blocks: FrozenDict[str, TextBlocks]) -> None:
+    if not isinstance(origin_text_blocks, FrozenDict):
+        raise ValueError(
+            f"Expected type `FrozenDict`, got {type(origin_text_blocks)=} {origin_text_blocks=}"
+        )
+    for blocks in origin_text_blocks.values():
+        if not isinstance(blocks, tuple):
```

Comment:
the signature says it should be a `TextBlocks` instance.. (and the signature is correct.)